pkg/golinters/gochecknoglobals.go

```diff
@@ -1,100 +1,30 @@
 package golinters
 
 import (
-	"fmt"
-	"go/ast"
-	"go/token"
-	"strings"
-	"sync"
+	"flag"
 
 	"golang.org/x/tools/go/analysis"
 
+	"4d63.com/gochecknoglobals/checknoglobals"
+
 	"github.com/golangci/golangci-lint/pkg/golinters/goanalysis"
-	"github.com/golangci/golangci-lint/pkg/lint/linter"
-	"github.com/golangci/golangci-lint/pkg/result"
 )
 
-const gochecknoglobalsName = "gochecknoglobals"
-
-//nolint:dupl
 func NewGochecknoglobals() *goanalysis.Linter {
-	var mu sync.Mutex
-	var resIssues []goanalysis.Issue
-
-	analyzer := &analysis.Analyzer{
-		Name: gochecknoglobalsName,
-		Doc:  goanalysis.TheOnlyanalyzerDoc,
-		Run: func(pass *analysis.Pass) (interface{}, error) {
-			var res []goanalysis.Issue
-			for _, file := range pass.Files {
-				fileIssues := checkFileForGlobals(file, pass.Fset)
-				for i := range fileIssues {
-					res = append(res, goanalysis.NewIssue(&fileIssues[i], pass))
-				}
-			}
-			if len(res) == 0 {
-				return nil, nil
-			}
+	// gochecknoglobals only lints test files if the `-t` flag is passed so we
+	// set up our own FlagSet and add it to the analyzer before running it. This
+	// can be turned of by using the regular golangci-lint flags such as
+	// `--tests` or `--skip-files`.
+	flags := flag.NewFlagSet("", flag.ContinueOnError)
+	flags.Bool("t", true, "Include tests")
```

Comment:
same as above point. Globally, we have include-test flag in golang.yaml config file.

Reply:
Global flag affects files that are passed to an Analyzer through `*analysis.Pass`
Analyzer could filter out test files as part of it's analysis (based on provided flags). `Gochecknoglobals` filters out tests by default.
I agree with @bombsimon that test filtering should be turned off (`-t` flag should be turned on). But this should be done by passing config, not by overriding flag set.

Reply:
yes, it's making sense to me, thanks :100: